### DISCUSSION TITLE
Alternative approach for MacOS implementation

### DISCUSSION THREAD

**Discussion by charankamarapu** [mitmproxy/mitmproxy#7199] [Q&A] [answered]
why is NETransparentProxyProvider and NEAppProxyTCPFlow used to send the traffic to mitmproxy..? Can't we just use NEPacketTunnelProvider and just change destinationIP and port..? Any specific reason for it..?

**Comment by mhils** [comment]
If I recall correctly, NEPacketTunnelProvider does not allow per-process filtering. @emanuele-em may remember more. :)

**Comment by emanuele-em** [comment, marked as answer]
@charankamarapu I reread my past notes and rediscovered that `NEPacketTunnelProvider` does not fill in the `sourceAppAuditToken`, the metadata in which the `PID` is stored; without the `PID`, we cannot figure out what the source process is and therefore cannot filter the packet based on the source application

**Reply by charankamarapu** [reply]
Thanks guys that cleared everything.
